bot/decorators.py

```diff
@@ -50,42 +51,34 @@ def predicate(ctx: Context):
     return commands.check(predicate)
 
 
-def with_role(*role_ids: int):
-    """
-    Returns True if the user has any one
-    of the roles in role_ids.
-    """
-
-    async def predicate(ctx: Context):
+def with_role(*role_ids: int) -> Callable:
+    """Returns True if the user has any one of the roles in role_ids."""
+    async def predicate(ctx: Context) -> bool:
+        """With role checker predicate."""
         return with_role_check(ctx, *role_ids)
     return commands.check(predicate)
 
 
-def without_role(*role_ids: int):
-    """
-    Returns True if the user does not have any
-    of the roles in role_ids.
-    """
-
-    async def predicate(ctx: Context):
+def without_role(*role_ids: int) -> Callable:
+    """Returns True if the user does not have any of the roles in role_ids."""
+    async def predicate(ctx: Context) -> bool:
         return without_role_check(ctx, *role_ids)
     return commands.check(predicate)
 
 
-def locked():
+def locked() -> Union[Callable, None]:
     """
     Allows the user to only run one instance of the decorated command at a time.
-    Subsequent calls to the command from the same author are
-    ignored until the command has completed invocation.
+
+    Subsequent calls to the command from the same author are ignored until the command has completed invocation.
 
     This decorator has to go before (below) the `command` decorator.
     """
-
-    def wrap(func):
+    def wrap(func: Callable) -> Union[Callable, None]:
         func.__locks = WeakValueDictionary()
 
         @wraps(func)
-        async def inner(self, ctx, *args, **kwargs):
+        async def inner(self: Callable, ctx: Context, *args, **kwargs) -> Union[Callable, None]:
```

Comment:
`inner` does not return a callable, but a union of `None` and the return value of the decorated original callable (`func`) in `return await func(...)`.

Since we don't strictly enforce commands to return `None`, by allowing the `return await ctx.send` pattern, I don't think we can make a reasonable assumption of what `func` will return, so I guess we could use `Optional[Any]` or something.

```suggestion
        async def inner(self: Callable, ctx: Context, *args, **kwargs) -> Optional[Any]:
```
(Note: if you like this suggestion, please also include `Any` and `Optional` in the `from typing import`; I've left a suggestion for that as well, to simplify that.)